DEVELOPER.md

```diff
@@ -4,72 +4,155 @@ The Terraform Provider for Power Platform extends Terraform's capabilities to al
+- Set or update your Git username and email:
 
 ```bash
 git config --global user.name "Your Name"
 git config --global user.email "your.email@address"
 ```
 
-Note: if you logging to docker container's shell outside the VS Code, in order to work with git repository, run the following commands:
+If accessing the container shell outside VS Code, run:
 
 ```bash
 export SSH_AUTH_SOCK=$(ls -t /tmp/vscode-ssh-auth* | head -1)
 export REMOTE_CONTAINERS_IPC=$(ls -t /tmp/vscode-remote-containers-ipc* | head -1)
 ```
 
-For more information about devcontainers, you can check out the [devcontainer documentation](https://code.visualstudio.com/docs/devcontainers/containers) and [sharing Git credentials with your container](https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials).
+For more details, see [sharing Git credentials with your container](https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials).
+
+### Using Terminal
+
+If you prefer to use you operating system terminal instead of VSCode you can run the following command:
+
+```bash
+docker exec -u vscode -w /workspaces/terraform-provider-power-platform -it <your_docker_container_name_goes_here> bash -c "exec bash"
+```
 
 ## Power Platform Prerequisites
 
-### Tenant
+Regardless of your chosen environment, you'll need access to a Power Platform tenant and appropriate credentials.
+
+### Tenant Setup
+
+Ensure you have access to a tenant where you can create and delete Power Platform environments and resources. Follow the [bootstrap readme](https://github.com/microsoft/power-platform-terraform-quickstarts/blob/main/bootstrap/README.md) from our quickstarts repository to set up your tenant.
+
+### Credentials and Authentication
 
-Developers should have access to a tenant where they can create and delete Power Platform environments and other resources and run acceptance test.
-You can setup access to you tenant using the following [bootstrap readme](https://github.com/microsoft/power-platform-terraform-quickstarts/blob/main/bootstrap/README.md) from our quickstarts repository.
+Refer to the [provider's user documentation](https://microsoft.github.io/terraform-provider-power-platform#authentication) for detailed instructions on authenticating to the provider.  When getting started we recommend authenticating with a user-context using `az login` command from az cli and `use_cli` provider configuration.
 
-### Credentials
+#### Service Principals
 
-See the [provider's user documentation](https://microsoft.github.io/terraform-provider-power-platform#authentication) on authenticating to the provider once your tenant is set up.
+While we recommend getting started using user-context authentication, most production scenarios will use service principals. Some APIs behave differently when called in a user-context vs service principal context so we highly recommend testing your contributions with a service principal.
```

Comment:
we should add one more sentence, that new feature should be tested not only with user login but also with sp login

Reply:
isn't that the next sentence? 😂